lonn plugins

Mnstrman06 opened the pull request:
TODO: arrow bubble texture support

Mnstrman06 committed:
lonn plugins

TODO: arrow bubble texture support

catapillie commented on Loenn/lang/en_gb.lang:
```diff
@@ -0,0 +1,43 @@
+# Mod Name
+mods.EverestYamlName.name=IsaGrabBag
```
You probably forgot to change `EverestYamlName`, which is supposed to match the mod's `everest.yaml` name. Whatever you write after the equal sign is what is shown in Lönn.
```suggestion
mods.IsaGrabBag.name=Isa's Grab Bag
```

catapillie reviewed (changes requested):
Please use `en_gb.lang` to give name to placements.

For example, let's say you are working on the `isaBag/arrowBubble` plugin. You have a placement for each direction, up, down, left and right. Instead of hardcoding the "human-readable" placements names into the plugin's code, give them a simple name like `"left"` for the left placements, `"down"` for the down one, etc... you get the idea.

You'll then want to associate these placement names to an actual readable name using the language file. So, in our example, it would be something like:
```txt
entities.isaBag/arrowBubble.placements.name.up=Arrow Bubble (Up)
entities.isaBag/arrowBubble.placements.name.down=Arrow Bubble (Down)
entities.isaBag/arrowBubble.placements.name.left=Arrow Bubble (Left)
entities.isaBag/arrowBubble.placements.name.right=Arrow Bubble (Right)
```

You don't need to mention `IsaGrabBag` in the placements names. Lönn will add it itself, given you defined the name in the language file. For example, in this case, it'd be:
```txt
mods.IsaGrabBag.name=Isa's Grab Bag
```
(Or, you know, you get to format it the way you want. I also like to put this at the top of the language file.)

bigkahuna443 commented on Loenn/entities/arrowBubble.lua:
```diff
@@ -0,0 +1,31 @@
+    }
+}
+
+#TODO: add texture based on direction
```
all the images are formatted as `booster{dir}00.png`, so you can use something like this:
`local texture = "isafriend/objects/booster/booster" .. dir .. "00"`

bigkahuna443 commented on Loenn/entities/arrowBubble.lua:
```diff
@@ -0,0 +1,31 @@
+arrowBubble.depth = -100
+arrowBubble.placements = {
+    {
+        name = "Arrow Bubble (Left, IsaGrabBag)",
```
to follow up on catapillie's comment, you would change this line to be `name = "left"` and then have `[...]placements.name.left = "Arrow Bubble (Left)"` in the en_gb.lang. check out some of the other legacy loenn plugin repos (or vanilla) for examples

Mnstrman06 committed:
Update arrowBubble.lua

Mnstrman06 committed:
Update baddyFollow.lua

Mnstrman06 committed:
Update dreamSpinnerFake.lua

bigkahuna443 commented on Loenn/entities/arrowBubble.lua:
```diff
@@ -0,0 +1,31 @@
+    {
+        name = "Arrow Bubble (Left, IsaGrabBag)",
+        data = {
+            direction = left
```
This doesn't load lol, looks like it's WIP though. I'd recommend making the `direction` field a non-editable dropdown with the directions. Looks like the code needs it to be lowercase which is annoying, I think there's a way to make it display separately though

Mnstrman06 committed:
Update dreamspinner.lua

Mnstrman06 committed:
Update pause.lua

Mnstrman06 committed:
Update rewind.lua

bigkahuna443 commented on Loenn/entities/rewind.lua:
```diff
@@ -0,0 +1,11 @@
+rewindCrystal.texture = "isafriend/objects/rewind/idle00"
+rewindCrystal.placements = {
+    name = "Rewind Crystal (IsaGrabBag)",
+    oneUse = true
```
should be in a `data` table

bigkahuna443 commented on Loenn/entities/waterBoost.lua:
```diff
@@ -0,0 +1,11 @@
+waterBoost.texture = "isafriend/helperimage"
+waterBoost.placements = {
+    name = "Water Boost Controller (IsaGrabBag)",
+    boostEnabled = true
```
should be in a `data` table

bigkahuna443 commented on Loenn/entities/zipline.lua:
```diff
@@ -0,0 +1,15 @@
+zipline.nodeLineRenderType = "fan"
+zipline.placements = {
+    name = "Zipline (IsaGrabBag)",
+    useStamina = true
```
should be in a `data` table

Mnstrman06 committed:
Update waterBoost.lua

Mnstrman06 committed:
Update zipline.lua

Mnstrman06 committed:
Update coreWindTrigger.lua

Mnstrman06 committed:
Update en_gb.lang

bigkahuna443 commented on Loenn/triggers/coreWindTrigger.lua:
```diff
@@ -0,0 +1,57 @@
+coreWindTrigger.fieldInformation = {
+    patternCold = {
+        options = {
+            "None",
```
use the `wind_patterns` enum for these

bigkahuna443 commented on Loenn/lang/en_gb.lang:
```diff
@@ -0,0 +1,43 @@
+mods.EverestYamlName.name=IsaGrabBag
+
+# Arrow Bubble
+entities.isaBag/arrowBubble.placements.name.default=arrow Bubble
```
please use consistent capitalization for these lol. so "arrow Bubble" -> "Arrow Bubble"

bigkahuna443 commented on Loenn/entities/zipline.lua:
```diff
@@ -0,0 +1,15 @@
+zipline.nodeTexture = "isafriend/objects/zipline/handle_end"
+zipline.nodeLimits = {0, -1}
+zipline.nodeVisibility = "always"
+zipline.nodeLineRenderType = "fan"
```
not sure what these choices are for nodes. use `{0, 2}` for limits, and it should be rendering the actual line like in the ahorn plugin

bigkahuna443 reviewed (changes requested):
Overall notes:
1. Please test plugins in-editor before making a PR, would've caught a lot of these.
2. Missing plugins for the Arrow Block, Corner Boost Block, and the Force Variant Trigger, not sure if you intended to add them or forgot. The latter two should not get placements but still need plugins

Mnstrman06 committed:
Update en_gb.lang

Mnstrman06 committed:
data table

Mnstrman06 committed:
Update rewind.lua

Mnstrman06 committed:
Update waterBoost.lua

Mnstrman06 committed:
Update zipline.lua

Mnstrman06 committed:
Update coreWindTrigger.lua

Mnstrman06 committed:
make direction field non-editable and add texture

Mnstrman06 committed:
Update en_gb.lang

Mnstrman06 committed:
Update arrowBubble.lua

Mnstrman06 committed:
finally fix arrow bubble

currently has a placeholder texture of the regular red bubble, i do not know enough about lua to make it change based on the direction of the bubble

Mnstrman06 committed:
Update en_gb.lang

Mnstrman06 committed:
cornerboost block

Mnstrman06 committed:
force variant

Mnstrman06 committed:
Add files via upload

Mnstrman06 committed:
various fixes

Mnstrman06 committed:
Update en_gb.lang

Mnstrman06 committed:
Update en_gb.lang

catapillie commented:
Is this still being worked on? I'm willing to review the code, and possibly improve the ports if necessary.

Mnstrman06 commented:
The ports are not finished, the remaining functions are things outside of my very limited skillset.
